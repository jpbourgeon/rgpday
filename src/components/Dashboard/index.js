import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import DashboardIcon from '@material-ui/icons/DashboardOutlined'
import AppBar from '../AppBar'
import Tile from '../Tile'
import Coconstruction from './images/coconstruction.jpg'
import Conference from './images/conference.jpg'
import GameBoard from './images/gameboard.jpg'
import Admin from './images/controlpanel.jpg'

const styles = theme => ({
  '@global': {
    '.grecaptcha-badge': {
      visibility: 'hidden'
    }
  },
  layout: {
    width: 'auto',
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(1100 + theme.spacing.unit * 3 * 2)]: {
      width: 1100,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  title: {
    paddingBottom: theme.spacing.unit * 3
  },
  titleIcon: {
    fontSize: '2.5rem',
    marginBottom: '-0.5rem'
  }
})

const DashboardComponent = props => {
  const { classes } = props
  return (
    <div className={classes.layout}>
      <AppBar />
      <main>
        <Typography variant='h4' color='inherit' gutterBottom className={classes.title}>
          <DashboardIcon className={classes.titleIcon} />&nbsp;Session
        </Typography>

        <Grid container spacing={40} className={''}>

          <Tile
            title='Administration du site'
            description=''
            image={Admin}
          />

          <Tile
            title='ConfÃ©rence'
            description=''
            image={Conference}
          />

          <Tile
            title='Serious game'
            description=''
            image={GameBoard}
          />

          <Tile
            title={`Plan d'action`}
            description=''
            image={Coconstruction}
          />

        </Grid>

      </main>
    </div>
  )
}

DashboardComponent.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(DashboardComponent)
