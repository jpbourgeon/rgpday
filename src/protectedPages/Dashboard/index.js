import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import Paper from '@material-ui/core/Paper'
import DashboardIcon from '@material-ui/icons/DashboardOutlined'
import Help from '@material-ui/icons/HelpOutline'
import Settings from '@material-ui/icons/SettingsOutlined'
import Tile from 'src/components/Tile'
import Coconstruction from './images/coconstruction.jpg'
import Conference from './images/conference.jpg'
import GameBoard from './images/gameboard.jpg'
import Admin from './images/controlpanel.jpg'
import Scenario from './images/scenario.jpg'
import Presentation from './images/presentation.jpg'

const styles = theme => ({
  '@global': {
    '.grecaptcha-badge': {
      visibility: 'hidden'
    }
  },
  layout: {
    width: 'auto',
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(1100 + theme.spacing.unit * 3 * 2)]: {
      width: 1100,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  paper: {
    backgroundSize: `100% 100%`,
    marginBottom: theme.spacing.unit * 6,
    padding: `0px ${theme.spacing.unit * 4}px ${theme.spacing.unit * 6}px ${theme.spacing.unit * 4}px`,
    textAlign: 'justify'
  },
  breadcrumb: {
    marginBottom: theme.spacing.unit
  },

  title: {
    paddingTop: theme.spacing.unit * 3
  },
  titleIcon: {
    fontSize: '2.5rem',
    marginBottom: '-0.5rem'
  }
})

const DashboardComponent = (props) => {
  const { classes, config } = props
  const renderSession = (!config.sessionId) ? null : (
    <Grid container spacing={40}>
      <Grid item xs={12}>
        <Typography variant='h4' color='inherit' gutterBottom className={classes.title}>
          <DashboardIcon className={classes.titleIcon} />&nbsp;Session {config.sessionId}
        </Typography>
      </Grid>
      <Tile
        title='Conférence'
        description=''
        image={Conference}
        to='/dashboard/presentation#/start'
      />
      <Tile
        title='Serious game'
        description=''
        image={GameBoard}
      />
      <Tile
        title={`Plan d'action`}
        description=''
        image={Coconstruction}
      />
    </Grid>
  )
  const renderAdmin = (!config.isAdmin) ? null : (
    <React.Fragment>
      <Grid container spacing={40}>
        <Grid item xs={12}>
          <Typography variant='h4' color='inherit' gutterBottom className={classes.title}>
            <Settings className={classes.titleIcon} />&nbsp;Administration
          </Typography>
        </Grid>
        <Tile
          title='Scénarios'
          description=''
          image={Scenario}
          to='./scenarios'
        />
        <Tile
          title='Présentations'
          description=''
          image={Presentation}
          to='./presentations'
        />
        <Tile
          title='Sessions'
          description=''
          image={Admin}
          to='./sessions'
        />
      </Grid>
      <Grid container spacing={40}>
        <Grid item xs={12}>
          <Typography variant='h4' color='inherit' gutterBottom className={classes.title}>
            <Help className={classes.titleIcon} />&nbsp;Aide
          </Typography>
        </Grid>
        <Tile
          title='Scénarios'
          description={`L'identifiant d'un scénario doit correspondre à un scénario disponible dans le code source.`}
        />
        <Tile
          title='Présentations'
          description={`L'identifiant d'une présentation doit correspondre à un scénario disponible dans le code source.`}
        />
        <Tile
          title='Sessions'
          description={(
            <React.Fragment>
              <span>1. Créer une session avec un identifiant de type [YYYY-ID-DE-SESSION]</span><br />
              <span>2. Créer un utilisateur vérifié dans la console AWS Cognito (no&#8209;reply@rgpday.com) avec pour nom d'utilisateur et pour mot de passe, l'identifiant de session créé à l'étape 1</span><br />
              <span>3. Valider le compte en se connectant sur le site dans les 7 jours suivants sa création</span>
            </React.Fragment>
          )}
        />
      </Grid>
    </React.Fragment>
  )
  return (
    <div className={classes.layout}>
      <main>
        <Paper className={classes.paper}>
          {renderSession}
          {renderAdmin}
        </Paper>
      </main>
    </div>
  )
}

DashboardComponent.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(DashboardComponent)
