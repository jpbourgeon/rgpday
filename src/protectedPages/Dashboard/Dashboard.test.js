/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import Dashboard from './'
import jssSerializer from '../../utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The Dashboard component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Dashboard />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
