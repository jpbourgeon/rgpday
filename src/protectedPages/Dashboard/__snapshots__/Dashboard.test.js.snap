// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Dashboard component should render correctly 1`] = `
<div
  className="layout-1"
>
  <main>
    <div
      className="container-4 40-30"
    >
      <div
        className="item-5 12-44"
      >
        <h4
          className="root-101 h4-116 colorInherit-130 gutterBottom-128 title-2"
        >
          <svg
            aria-hidden="true"
            className="root-137 titleIcon-3"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <g>
              <path
                d="M19 5v2h-4V5h4M9 5v6H5V5h4m10 8v6h-4v-6h4M9 17v2H5v-2h4M21 3h-8v6h8V3zM11 3H3v10h8V3zm10 8h-8v10h8V11zm-10 4H3v6h8v-6z"
              />
            </g>
          </svg>
           Session
        </h4>
      </div>
      <div
        className="item-5 12-44 6-66"
      >
        <div
          className="root-150 elevation1-153 rounded-151 root-149 card-146"
        >
          <div
            className="cardDetails-147"
          >
            <div
              className="root-177"
            >
              <h5
                className="root-101 h5-117"
              >
                Conférence
              </h5>
              <p
                className="root-101 subtitle1-119 paragraph-129"
              >
                
              </p>
            </div>
          </div>
          <div
            className="root-178 cardMedia-148"
            style={
              Object {
                "backgroundImage": "url(\\"conference.jpg\\")",
              }
            }
            title="Conférence"
          />
        </div>
      </div>
      <div
        className="item-5 12-44 6-66"
      >
        <div
          className="root-150 elevation1-153 rounded-151 root-149 card-146"
        >
          <div
            className="cardDetails-147"
          >
            <div
              className="root-177"
            >
              <h5
                className="root-101 h5-117"
              >
                Serious game
              </h5>
              <p
                className="root-101 subtitle1-119 paragraph-129"
              >
                
              </p>
            </div>
          </div>
          <div
            className="root-178 cardMedia-148"
            style={
              Object {
                "backgroundImage": "url(\\"gameboard.jpg\\")",
              }
            }
            title="Serious game"
          />
        </div>
      </div>
      <div
        className="item-5 12-44 6-66"
      >
        <div
          className="root-150 elevation1-153 rounded-151 root-149 card-146"
        >
          <div
            className="cardDetails-147"
          >
            <div
              className="root-177"
            >
              <h5
                className="root-101 h5-117"
              >
                Plan d'action
              </h5>
              <p
                className="root-101 subtitle1-119 paragraph-129"
              >
                
              </p>
            </div>
          </div>
          <div
            className="root-178 cardMedia-148"
            style={
              Object {
                "backgroundImage": "url(\\"coconstruction.jpg\\")",
              }
            }
            title="Plan d'action"
          />
        </div>
      </div>
    </div>
    <div
      className="container-4 40-30"
    >
      <div
        className="item-5 12-44"
      >
        <h4
          className="root-101 h4-116 colorInherit-130 gutterBottom-128 title-2"
        >
          <svg
            aria-hidden="true"
            className="root-137 titleIcon-3"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <g>
              <path
                d="M19 5v2h-4V5h4M9 5v6H5V5h4m10 8v6h-4v-6h4M9 17v2H5v-2h4M21 3h-8v6h8V3zM11 3H3v10h8V3zm10 8h-8v10h8V11zm-10 4H3v6h8v-6z"
              />
            </g>
          </svg>
           Administration
        </h4>
      </div>
      <div
        className="item-5 12-44 6-66"
      >
        <div
          className="root-150 elevation1-153 rounded-151 root-149 card-146"
        >
          <div
            className="cardDetails-147"
          >
            <div
              className="root-177"
            >
              <h5
                className="root-101 h5-117"
              >
                <a
                  className="root-101 colorPrimary-131 inline-136 root-180 underlineHover-182"
                  href="//scenarios"
                  onClick={[Function]}
                >
                  Scenarios
                </a>
              </h5>
              <p
                className="root-101 subtitle1-119 paragraph-129"
              >
                
              </p>
            </div>
          </div>
          <div
            className="root-178 cardMedia-148"
            style={
              Object {
                "backgroundImage": "url(\\"scenario.jpg\\")",
              }
            }
            title={
              <WithStyles(Link)
                color="primary"
                component={[Function]}
                to="./scenarios"
              >
                Scenarios
              </WithStyles(Link)>
            }
          />
        </div>
      </div>
      <div
        className="item-5 12-44 6-66"
      >
        <div
          className="root-150 elevation1-153 rounded-151 root-149 card-146"
        >
          <div
            className="cardDetails-147"
          >
            <div
              className="root-177"
            >
              <h5
                className="root-101 h5-117"
              >
                <a
                  className="root-101 colorPrimary-131 inline-136 root-180 underlineHover-182"
                  href="//sessions"
                  onClick={[Function]}
                >
                  Sessions
                </a>
              </h5>
              <p
                className="root-101 subtitle1-119 paragraph-129"
              >
                
              </p>
            </div>
          </div>
          <div
            className="root-178 cardMedia-148"
            style={
              Object {
                "backgroundImage": "url(\\"controlpanel.jpg\\")",
              }
            }
            title={
              <WithStyles(Link)
                color="primary"
                component={[Function]}
                to="./sessions"
              >
                Sessions
              </WithStyles(Link)>
            }
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
